### PR TITLE
Move process related fields to their own namespace

### PR DESCRIPTION
**Pull request by roncohen**
This affects the intake API as well as the output documents.

This puts us on the path to align better with other elastic projects in the future.

### CODE REVIEW

File: _meta/fields.common.yml [hunk 1]
```diff
@@ -141,6 +141,21 @@
               type: keyword
               description: >
                 The platform of the system the event was recorded on.
+        - name: process
+          type: group
+          description: >
+            Information pertainin to the running process where the data was collected
```
Comment (graphaelli) [inline _meta/fields.common.yml]: typo in `pertainin`

File: docs/spec/errors/payload.json [hunk 2]
```diff
@@ -1,13 +1,16 @@
 {
     "$schema": "http://json-schema.org/draft-04/schema#",
     "$id": "docs/spec/errors/wrapper.json",
```
Comment (graphaelli) [inline docs/spec/errors/payload.json]: while you're at it, `payload.json`

File: processor/error/schema.go [hunk 3]
```diff
@@ -7,8 +7,8 @@ func Schema() string {
 var errorSchema = `{
     "$schema": "http://json-schema.org/draft-04/schema#",
     "$id": "docs/spec/errors/wrapper.json",
```
Comment (graphaelli) [inline processor/error/schema.go]: again `payload.json`

**Comment by roncohen** [comment]
thanks @graphaelli ! Fixed nits and rebased to resolve conflicts.

**Comment by roncohen** [comment]
Breaking change: @elastic/apm-ui  @elastic/apm-agent-devs

**Comment by beniwohli** [comment]
@roncohen when is this supposed to land? If 6.2, can you add it to the 6.2 checklist?

**Comment by roncohen** [comment]
@beniwohli done!

File: docs/spec/process.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "$id": "doc/spec/process.json",
+  "title": "Process",
+  "type": ["object", "null"],
+  "properties": {
+      "pid": {
+          "description": "Process ID of the service",
+          "type": ["number", "null"]
```
Comment (simitt) [inline docs/spec/process.json]: why is `null` allowed when it is a required field?

File: model/process.go [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package model
+
+import (
+	"github.com/elastic/apm-server/utility"
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type Process struct {
+	Pid   *int
```
Comment (simitt) [inline model/process.go]: When `pid` is required you could use `int` instead of `*int` here.

File: processor/transaction/event.go [hunk 6]
```diff
@@ -34,14 +34,20 @@ func (t *Event) Transform() common.MapStr {
...
+		{Key: "context", Apply: func() common.MapStr { return t.Context }},
+		{Key: "context.service", Apply: pa.Service.Transform},
+		{Key: "context.system", Apply: pa.System.Transform},
+	}
+
+	if pa.Process != nil {
+		mapping = append(mapping, m.DocMapping{
+			Key: "context.process", Apply: pa.Process.Transform,
+		})
+	}
```
Comment (simitt) [inline processor/transaction/event.go]: Why do you handle this differently than for `errors` here https://github.com/elastic/apm-server/pull/445/files#diff-a163837d2241a4bad19d7e4baa4c617bR62?

I suggest to rather add a nil check when processing the mappings, as e.g. `system` is also not a required field.

File: processor/error/package_tests/TestProcessErrorNullValues.approved.json [hunk 7]
```diff
@@ -3,6 +3,7 @@
         {
             "@timestamp": "2017-05-09T15:04:05.999999Z",
             "context": {
+                "process": {},
```
Comment (simitt) [inline processor/error/package_tests/TestProcessErrorNullValues.approved.json]: I think `process` should be able to be `null`, so this should be checked here. 
If `process` is given, it should however have a `pid`, at least this is required in the spec above. If you remove the `null` type in the spec, sending an empty hash for `process` should not be possible any longer.

File: processor/transaction/package_tests/TestProcessTransactionNullValues.approved.json [hunk 8]
```diff
@@ -3,6 +3,7 @@
         {
             "@timestamp": "2017-05-30T18:53:27.154Z",
             "context": {
+                "process": {},
```
Comment (simitt) [inline processor/transaction/package_tests/TestProcessTransactionNullValues.approved.json]: same comment as above - empty `process` does not reflect spec, where `pid` is required.

File: tests/data/valid/transaction/null_values.json [hunk 9]
```diff
@@ -22,6 +22,11 @@
         "architecture": null,
         "platform": null
     },
+    "process": {
+        "pid": null,
```
Comment (simitt) [inline tests/data/valid/transaction/null_values.json]: again - this doesn't make sense to me - either `process.pid` is required, or it is allowed to be null.

File: processor/transaction/event.go (hunk 6 above)
Comment (roncohen) [reply processor/transaction/event.go]: oversight on my part, good catch

**Comment by roncohen** [comment]
great feedback. Thanks @simitt !

**Comment by simitt** [comment]
@roncohen some tests are failing, apart from that LGTM

**Comment by beniwohli** [comment]
Implemented for Python in https://github.com/elastic/apm-agent-python/pull/125

**Comment by makwarth** [comment]
@roncohen could you elaborate on the UI implications?

**Comment by roncohen** [comment]
@makwarth yes. The data is moved around in the error document and the transactions document:

Transaction: https://github.com/elastic/apm-server/pull/445/files#diff-f63f8f194850ee66420e85c23183a538

Error: https://github.com/elastic/apm-server/pull/445/files#diff-ce810586885ee0e452b0c394d2d61ee0

**Pull request merged**